opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/__init__.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+from typing import Sequence, Tuple
+
+from .. import Metric
+
+
+class MetricsExportResult(Enum):
+    SUCCESS = 0
+    FAILED_RETRYABLE = 1
+    FAILED_NOT_RETRYABLE = 2
+
+
+class MetricsExporter:
+    """Interface for exporting metrics.
+
+    Interface to be implemented by services that want to export recorded in
+    its own format.
+    """
+
+    def export(
+        self, metric_tuples: Sequence[Tuple[Metric, Sequence[str]]]
+    ) -> "MetricsExportResult":
+        """Exports a batch of telemetry data.
+
+        Args:
+            metric_tuples: A sequence of metric pairs. A metric pair consists
+                of a `Metric` and a sequence of strings. The sequence of
+                 strings will be used to get the corresponding `MetricHandle`
+                from the `Metric` to import.
+
+        Returns:
+            The result of the export
+        """
+
+    def shutdown(self) -> None:
+        """Shuts down the exporter.
+
+        Called when the SDK is shut down.
+        """
+
+
+class ConsoleMetricsExporter(MetricsExporter):
+    """Implementation of :class:`MetricsExporter` that prints metric handles
+    to the console.
+
+    This class can be used for diagnostic purposes. It prints the exported
+    metric handles to the console STDOUT.
+    """
+
+    def export(
+        self, metric_tuples: Sequence[Tuple[Metric, Sequence[str]]]
+    ) -> "MetricsExportResult":
+        for metric_tuple in metric_tuples:
+            handle = metric_tuple[0].get_handle(metric_tuple[1])
+            print(
```

Comment:
could we have this be emitter to a logger vs printed to stdout? I'm thinking it would be nice to redirect this stream to something else (e.g. some streaming service), and loghandlers are a great way to redirect that stream.

conversely I could imagine maybe passing a file object? and then write to it?

Reply:
I believe the main idea of the console exporter is for simple debugging purposes, so simply printing out to the console could be enough. Although there may be cases where the outputs get overloaded, in which case redirecting to a file would allow easiness for debugging. I think we can revisit this in a separate issue.

Reply:
IIRC we talked about this and decided it made sense to have separate console and logging exporters so users would have an impossible-to-screw-up way to test that telemetry is working. We considered implementing the console logger as a special case of the logging exporter but decided against it because it would be possible to modify the logger (or handler directing logs to stdout) via config.